pkg/cmd/miscellaneous.go

```diff
@@ -350,17 +376,69 @@ func getRole() string {
 }
 
 /*
-getTechnicalID returns the Technical Id, which used as clustername of the shoot cluster
+getTargetMapInfo retun garden,project,seed,shoot,shootTechnicalID to global targetInfo
 */
-func getTechnicalID() string {
+func getTargetMapInfo() {
+	if len(targetInfo) > 0 {
+		return
+	}
 	var target Target
 	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		targetInfo[string(t.Kind)] = string(t.Name)
+	}
 
-	gardenClientset, err := target.GardenerClient()
-	checkError(err)
-	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
-	checkError(err)
-	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
+	shoot, err := getShootObject()
 	checkError(err)
-	return shoot.Status.TechnicalID
+	targetInfo["shootTechnicalID"] = shoot.Status.TechnicalID
+
+	// user for case garden-project-shoot add [seed] = seedName
+	if isTargeted("project", "shoot") {
+		targetInfo["seed"] = *shoot.Spec.SeedName
+	}
+}
+
+/*
+lookup Target Kind ("kind value") return Target Name "name value" from target file ~/.garden/sessions/plantingSession/target
+*/
+func getTargetName(Kind string) (string, error) {
+	var target Target
+	ReadTarget(pathTarget, &target)
+	for _, t := range target.Stack() {
+		if string(t.Kind) == Kind {
+			return string(t.Name), nil
+		}
+	}
+	return "", errors.New("Kind:" + Kind + "not found from ~/.garden/sessions/plantingSession/target")
+}
+
+/*
+check if target Kind is exist in target file ~/.garden/sessions/plantingSession/target
+*/
+func isTargeted(args ...string) bool {
+	var target Target
+	ReadTarget(pathTarget, &target)
+
+	targetMap := make(map[string]interface{})
+	for _, t := range target.Stack() {
+		targetMap[string(t.Kind)] = string(t.Name)
+	}
+
+	for _, t := range args {
+		if _, ok := targetMap[t]; ok {
+		} else {
+			return false
+		}
+	}
+
+	return true
+}
+
+//getFromTargetInfo validation value from global map targetInfo garden/project/shoot/seed/shootTechnicalID/....
+func getFromTargetInfo(key string) string {
+	value := targetInfo[key]
+	if value == "" {
+		fmt.Printf("value %s not found in targetInfo\n", key)
+	}
+	return value
 }
```

Comment:
@tedteng the `getFromTargetInfo` function does not include proper error handling! Either os.exit or return an error.
I have included this even in my example but you seem to have removed this for reasons not known to me https://github.com/gardener/gardenctl/pull/319#discussion_r493311340

Reply:
missing that part interrupted by other issues today.  will add it

Reply:
done